Rubberduck.Core/UI/UnitTesting/TestExplorerViewModel.cs

```diff
@@ -341,6 +388,28 @@ private void ExecuteResetResultsCommand(object parameter)
             Tests.Refresh();
         }
 
+        private void ExecuteToggleTestAnnotationCommand(object parameter)
+        {
+            var selectedTestMethodViewModel = (TestMethodViewModel)SelectedItem;
+            var testMethod = selectedTestMethodViewModel.Method;
+
+            var annotationUpdater = new AnnotationUpdater();
+            var rewriteSession = RewritingManager.CheckOutCodePaneSession();
+
+            if (testMethod.IsIgnored)
+            {
+                var ignoreTestAnnotation = testMethod.Declaration.Annotations.First(iannotation => iannotation.AnnotationType == Parsing.Annotations.AnnotationType.IgnoreTest);
```

Comment:
Is it possible for `Annotations` to be empty here? If it’s guaranteed to have at least one element, this is fine.

Reply:
While it is possible I don't believe it will/should be. I'm working under the assumption that `@TestMethod` will always there. With the ability to ignore a test I can't envision a reason as to why it would need to be removed.

Reply:
The assumption that a `TestMethod` will always have a `@TestMethod` annotation is fundamentally correct; in 1.0 the annotation was optional and we could discover tests by name (e.g. any public parameterless method in a standard module with a name that starts (ends?) with "Test", is a test), I don't think we will be rolling that back.

That said the use of `.First` instead of `.Single` is legit (there *could* be illegal duplicate annotations of that type), but IMO it warrants a comment e.g. `// ignore illegal duplicate annotations`.